library/Icingadb/Web/Controller.php

```diff
@@ -453,10 +468,14 @@ function (ViewModeSwitcher $viewModeSwitcher) use ($user, $preferredModes, $pagi
             }
         )->handleRequest(ServerRequest::fromGlobals());
 
-        if ($viewModeSwitcher->getViewMode() === 'minimal') {
+        $previousMode = $session->get('previous_mode');
```

Comment:
What does this fix? Or why is it required for this fix?

Reply:
Because when switching from minimal to the other modes and we use the vertical pagination, the default page limit remains unchanged and we don't have to recalculate the page number.